### ISSUE TITLE
Add support pre-authentication using Apache

### ISSUE THREAD

**Issue by GastonGonzalez** [headwirecom/peregrine-cms#322]
For Peregrine instances running behind a reverse proxy (i.e. Apache HTTPD + mod_proxy) there may be a worthwhile use case that allows the web tier to perform user authentication and authorization. For example, a module such as [mod_auth_openidc](https://github.com/zmartzone/mod_auth_openidc) can be deployed to Apache HTTPD and perform the actual OpenID Connect authentication, then forward the authenticated user as a request header to Peregrine. This approach would allow the Apache HTTPD server to handle multiple, different types of authentication and pass authenticated credentials for Peregrine.

**Comment by GastonGonzalez** [comment]
An initial branch (issue/332) has been created that includes a version of the Sling9 launchpad with an External Header Login Module. Note: Since this is still in early development, a local Maven build is required for the External Header Login Module (See https://github.com/peregrine-cms/peregrinecms-com-peregrine-sling-auth-header).

**Comment by cmrockwell** [comment]
Hi Gaston

I tried your code, and it worked well for me. At least as far as the processes discussed in the README. We discussed pre-authenticated / SSO options a little bit in the last community hangout. SSO using header security works for some user-cases. For other use-cases an application level authentication provider may be more suitable. 

Header Security
* Pros: Simpler to implement with more options (mod_cosign, mod_oidc, etc)  
* Cons: Security depends on preventing bypass of the web server (httpd), I think that all users need to be created and managed

Sling Authentication Handler
* Pros: User management service can synchronize group membership, so only ACL's on groups need management, users are automatically created, user attributes may be synchronized to the Sling user based on the what the identity provider releases, SAML assertions and signed, verified and encrypted  
* Cons: It is more complicated to configure.

I developed a SAML2 authentication handler for use with Peregrine CMS.  if you had any available cycles to review the code and offer any comments or feedback, then it would be really well appreciated. 

https://github.com/apache/sling-whiteboard/pull/51
